engine/lib/phases/phase_functionalize_loops.ml

```diff
@@ -129,29 +140,45 @@ struct
         Some (ChunksExact { size; slice })
       else None
 
-    let fn_args_of_iterator (it : iterator) :
-        (Concrete_ident.name * B.expr list * B.ty) option =
+    let fn_args_of_iterator (has_cf : bool) (has_return : bool) (it : iterator)
```

Comment:
I feel like those boolean flags are not very nice, we only have three states: no control flow, only returns, both returns and breaks. But if we want to go from those two bools to this enum, we need to lift that to the AST or something